matrix_sdk_crypto/src/memory_stores.rs

```diff
@@ -175,7 +175,9 @@ impl DeviceStore {
         let device_map = self.entries.get_mut(&user_id).unwrap();
 
         device_map
-            .insert(device.device_id().to_owned(), device)
+            // TODO this is ok if this is for sure a valid device_id otherwise
+            // Box::<DeviceId>::try_from(&str) is the validated version
```

Comment:
`DeviceId` is just an alias for `str`, there is no validation. The only `TryFrom` conversion there is is the blanked impl for all `From` types.